src/Symfony/Component/Finder/Iterator/ExcludeDirectoryFilterIterator.php

```diff
@@ -70,6 +83,14 @@ public function accept(): bool
             return !preg_match($this->excludedPattern, $path);
         }
 
+        if ($this->isRecursive && $this->isDir()) {
```

Comment:
psalm is wong here, see `$this->isDir()` used a few lines above, the code is correct